facebook/facebook-json.c

```diff
@@ -60,8 +62,15 @@ json_value *fb_json_new(const gchar *data, gsize length, GError **err)
         return json;
     }
 
+    /* Ensure it's null-terminated before passing it to g_strescape() */
+    dstr = g_strndup(data, MIN(length, 400));
+    escaped = g_strescape(dstr, "\"");
+
     g_set_error(err, FB_JSON_ERROR, FB_JSON_ERROR_PARSER,
-                "Parser: %s", estr);
+                "Parser: %s\nJSON len=%zd: %s", estr, length, escaped);
```

Comment:
It does look weird (just like in the example of the description), but it's mostly like that to have space for more characters.